### PR TITLE
[python] Remove `pyarrow` pin on MacOS

### PR DESCRIPTION
**Pull request by johnkerl**
Long-awaited work coming to fruition:

* https://github.com/single-cell-data/TileDB-SOMA/issues/1849
* https://github.com/single-cell-data/TileDB-SOMA/pull/2692
* https://github.com/single-cell-data/TileDB-SOMA/pull/2734
* https://github.com/TileDB-Inc/TileDB/pull/5223 went into [core 2.26](https://github.com/TileDB-Inc/TileDB/releases/tag/2.26.0), making this possible

See also https://github.com/single-cell-data/TileDB-SOMA/issues/2999 [[sc-53002]](https://app.shortcut.com/tiledb-inc/story/53002/update-projects-to-tiledb-2-26)

### CODE REVIEW

File: apis/python/src/tiledbsoma/__init__.py [hunk 1]
```diff
@@ -150,10 +150,11 @@ def _new_shape_feature_flag_enabled() -> bool:
 from somacore import AxisColumnNames, AxisQuery, ExperimentAxisQuery
 from somacore.options import ResultOrder
 
-# TODO: once we no longer support Python 3.7, remove this and pin to pyarrow >= 14.0.1
-# https://github.com/single-cell-data/TileDB-SOMA/issues/1926
+# This is important since we need to do the above dll/dylib/so business
+# _before_ imports, but, ruff will tell us that imports need to be
+# at the top of the file:
+#
```
Comment (johnkerl) [inline apis/python/src/tiledbsoma/__init__.py:156]: Link for #3001

**Review by teo-tsirpanis — APPROVED** [review]

**Comment by johnkerl** [comment]
@teo-tsirpanis has approved; not sure what happened to his approval but it's past EOD in his timezone

**Comment by teo-tsirpanis** [comment]
I don't have write permissions to this repository.

**Review by bkmartinjr — APPROVED** [review]
LGTM

**Comment by johnkerl** [comment]
Ah, thanks @teo-tsirpanis ! :)

**Pull request merged**